platform/android/src/run_loop.cpp

```diff
@@ -217,8 +217,10 @@ LOOP_HANDLE RunLoop::getLoopHandle() {
 }
 
 void RunLoop::push(std::shared_ptr<WorkTask> task) {
-    withMutex([&] { queue.push(std::move(task)); });
-    impl->wake();
+    withMutex([&] {
+        queue.push(std::move(task));
+        impl->wake();
+    });
```

Comment:
Now that we lock the entire function, we can also remove the `withMutex` function altogether and just move the `std::lock_guard` into every function.

Reply:
:+1:

Reply:
`withMutex` is also used in `process`, and I think there's a readability win in using the same syntax to mark all locked sections in the code. If we were to stop using `withMutex` in `push`, I think it would also make sense to change the call in `process` to something like:

    ...
    {
      std::lock_guard<std::mutex> lock(mutex);
      queue_.swap(queue);
    }
    ...

Between those two options... 🤷‍♂️ ? Doesn't make a big difference to me, but at some point we thought `withMutex` was a clearer way to mark the locked section of code.